### PR TITLE
[runtime-security] add runtime container test and json schema validation

### PR DESCRIPTION
**Pull request by safchain**
### What does this PR do?

Validate container event post agent start and introduce json schema validation for exec event.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Write there any instructions and details that should be tested during the QA.

### CODE REVIEW

File: pkg/security/probe/model.go [hunk 1]
```diff
@@ -470,12 +491,12 @@ func NewProcessCacheEntry() *model.ProcessCacheEntry {
 }
 
 // ResolveProcessContextUser resolves the user id of the process to a username
-func (ev *Event) ResolveProcessContextUser(p *model.ProcessContext) string {
+func (ev *Event) ResolveProcessContextUser(ctx *eval.Context, p *model.ProcessContext) string {
```
Comment (lebauce) [inline pkg/security/probe/model.go:494]: Doesn't seem to be used

File: pkg/security/probe/model.go [hunk 2]
```diff
@@ -470,12 +491,12 @@ func NewProcessCacheEntry() *model.ProcessCacheEntry {
...
 	return ev.resolvers.ResolveProcessContextUser(p)
 }
 
 // ResolveProcessContextGroup resolves the group id of the process to a group name
-func (ev *Event) ResolveProcessContextGroup(p *model.ProcessContext) string {
+func (ev *Event) ResolveProcessContextGroup(ctx *eval.Context, p *model.ProcessContext) string {
```
Comment (lebauce) [inline pkg/security/probe/model.go:499]: ditto

File: pkg/security/secl/eval/context.go [hunk 3]
```diff
@@ -17,6 +17,9 @@ type Context struct {
 
 	// cache available across all the evaluations
 	Cache map[string]unsafe.Pointer
+
+	// state can be used to store custom state
+	State int
```
Comment (lebauce) [inline pkg/security/secl/eval/context.go:22]: not used

**Review by lebauce — APPROVED** [review]

**Review by Gui774ume — APPROVED** [review]
lgtm

**Pull request merged**